src/unwinder/arch/riscv32.rs

```diff
@@ -133,6 +136,8 @@ macro_rules! code {
     };
     (restore_fp) => {
         "
+        .option push     // Work around Rust issue #80608
+        .option arch, +d // Work around Rust issue #80608
```

Comment:
I think this isn't necessary, this is only used in normal asm which doesn't have the global_asm/naked_asm bug.

Reply:
It does seem that way. Let me remove that (and in riscv64) and see if it passes.